src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -10944,6 +10944,10 @@ private BoundConditionalAccess BindConditionalAccessExpression(ConditionalAccess
 
             var conditionalAccessBinder = new BinderWithConditionalReceiver(this, receiver);
             var access = conditionalAccessBinder.BindValue(node.WhenNotNull, diagnostics, BindValueKind.RValue);
+            if (access.Syntax is AssignmentExpressionSyntax)
+            {
+                CheckFeatureAvailability(node.WhenNotNull, MessageID.IDS_FeatureNullConditionalAssignment, diagnostics);
```

Comment:
Consider placing the diagnostics just on the `=`.  It's generally nicer than a big squiggle on something large.

Reply:
good idea!